prime_factor/prime_factor.c

```diff
@@ -0,0 +1,43 @@
+#include <stdio.h>
+#include <string.h> // for memset
+#include <stdbool.h> // for bool type
+#include <math.h>
+
+int prime_factors(int num, int *primes) {
+    int cnt = 0; // index to the last location in the array
+    bool flag[num];
+    memset(flag, true, sizeof(flag));
+    flag[0] = false;
+    flag[1] = false;
+    for (int i = 2; i*i < num; ++i) {
+        if (flag[i]) {
+            if (num % i == 0) {
+                for (int j = 2 * i; j < num; j += i) {
```

Comment:
But them it gives wrong answer as 1562 has a prime factor 71 which is greater than sqrt(num)